tutel/gates/top.py

```diff
@@ -4,12 +4,12 @@
 import torch
 
 class LinearTopKGate(torch.nn.Module):
-    def __init__(self, model_dim, num_global_experts, k=1, fp32_gate=False, **options):
+    def __init__(self, model_dim, num_global_experts, k=1, fp32_gate=False, device=None, **options):
         super().__init__()
         try:
-            self.wg = torch.nn.Linear(model_dim, num_global_experts, bias=False, dtype=torch.float32 if fp32_gate else None)
+            self.wg = torch.nn.Linear(model_dim, num_global_experts, bias=False, device=device, dtype=torch.float32 if fp32_gate else None)
         except:
-            self.wg = torch.nn.Linear(model_dim, num_global_experts, bias=False)
+            self.wg = torch.nn.Linear(model_dim, num_global_experts, bias=False, device=device)
         self.top_k = min(num_global_experts, int(k))
```

Comment:
What's the purpose of letting gate layer to initialize with specific devices, while expert layer still uses layer.to(device)?

Reply:
How about using `def to(self, *args, **kwargs)` as well?

Reply:
things like `device` and `dtype` are part of all torch layers which have parameters (eg [torch.nn.Linear](https://pytorch.org/docs/stable/generated/torch.nn.Linear.html)).
I like to propagate them.
When you start getting fancy with how you initializing things when using different dist strategies, propagating these args can be useful.

Reply:
I should probably propagate `dtype`.
and also both of these args to the expert layers.